CMakeLists.txt

```diff
@@ -486,7 +486,28 @@ elseif (WIN32 AND NOT MINGW)
 		endif()
 
 		#find pthread library
-		find_library(PTHREAD_LIBRARY NAMES pthread pthread_dll pthread_lib HINTS C:/pthread-win32/lib C:/pthread-win64/lib ${PTHREAD_PACKAGE_LIB_HINT})
+		set(PTHREAD_LIB_SUFFIX "")
+		if (ENABLE_DEBUG)
+			set(PTHREAD_LIB_SUFFIX "d")
+		endif ()
+
+		set(PTHREAD_COMPILER_FLAG "")
+		if (MICROSOFT)
+			set(PTHREAD_COMPILER_FLAG "V")
+		elseif (MINGW)
+			set(PTHREAD_COMPILER_FLAG "G")
+		endif ()
+
+		set(PTHREAD_W32_LIBRARY 
```

Comment:
Couldn't this be better encoded with some kind of FOR loop?

Reply:
Kinda:
```
FOREACH(EXHAND C CE SE)
   FOREACH(COMPAT 1 2)
    LIST(APPEND PTHREAD_W32_LIBRARY pthread${PTHREAD_COMPILER_FLAGS}${EXHAND}${PTHREAD_LIB_SUFFIX}${COMPAT})
...
```

Reply:
good idea that I didn't come up (for the lack of cmake skill)